src/API/Reports/Coupons/DataStore.php

```diff
@@ -57,11 +57,11 @@ class DataStore extends ReportsDataStore implements DataStoreInterface {
 	 */
 	protected function assign_report_columns() {
 		$table_name           = self::get_db_table_name();
-		$this->report_columns = array(
+		$this->report_columns =  apply_filters('wc_analytics_' . $this->context . '_report_columns', array(
```

Comment:
Instead of applying this filter manually in each report, I would much rather see a filter that catches assigning of report columns.  I.e., in https://github.com/woocommerce/woocommerce-admin/blob/cf23784df76838738a60b6bdbda9bc3c63e5ed6c/src/API/Reports/DataStore.php#L101